lib/OpenQA/WebAPI/Auth/OAuth2.pm

```diff
@@ -78,6 +80,7 @@ sub update_user ($controller, $main_config, $provider_config, $data) {
         return $controller->render(text => $msg, status => 403);    # return always 403 for consistency
     }
     my $details = $tx->res->json;
+    log_debug("OAuth2 user provider returned: " . Dumper($details));
```

Comment:
If this information is only necessary in case of the error message "User data returned by OAuth2 provider is insufficient", then I guess this line should be moved to inside the next `if`

Reply:
Are there error paths later that could be triggered by bad user info? If no, then yes let's move it inside the `if`. That might also help with @Martchus comment (no errors = nothing logged)

Reply:
To answer my own question, the only codepath using $details is in the same function, and has no special error handling so should already log a detailed message. I will make this change.